### PR TITLE
feat: add isNetworkEnabled function

### PR DESCRIPTION
**Pull request by salimtb**
## Explanation

The NetworkEnablementController was missing a public method to check whether a specific network is enabled. This functionality is needed by consumers to query the enablement state of networks without having to access the internal state structure directly.
The solution adds the isNetworkEnabled method which:

- Accepts both Hex chain IDs (e.g., '0x1') and CAIP chain IDs (e.g., 'eip155:1')
- Returns a boolean indicating whether the network is enabled
- Gracefully handles edge cases like non-existent networks or namespaces by returning false
- Uses the existing deriveKeys utility to normalize different chain ID formats
- Leverages optional chaining to prevent errors when accessing undefined namespace properties

Additionally, this PR adds Palm network and HypeEVM network to the list of popular networks, expanding the supported network ecosystem.
The method includes comprehensive test coverage to ensure reliability across different blockchain namespaces (EVM, Solana, etc.) and various edge cases including invalid inputs and non-existent networks.

## Checklist

- [ ] I've updated the test suite for new or updated code as appropriate
- [ ] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [ ] I've communicated my changes to consumers by [updating changelogs for packages I've changed](https://github.com/MetaMask/core/tree/main/docs/contributing.md#updating-changelogs), highlighting breaking changes as necessary
- [ ] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/network-enablement-controller/src/NetworkEnablementController.ts [hunk 1]
```diff
@@ -247,6 +247,20 @@ export class NetworkEnablementController extends BaseController<
...
+   * @param chainId - The chain ID of the network to check. Can be either:
+   * - A Hex string (e.g., '0x1' for Ethereum mainnet) for EVM networks
+   * - A CAIP-2 chain ID (e.g., 'eip155:1' for Ethereum mainnet, 'solana:mainnet' for Solana)
+   * @returns True if the network is enabled, false otherwise
+   */
+  isNetworkEnabled(chainId: Hex | CaipChainId): boolean {
+    const derivedKeys = deriveKeys(chainId);
+    const { namespace, storageKey } = derivedKeys;
+    return this.state.enabledNetworkMap[namespace]?.[storageKey] ?? false;
+  }
```
Comment (Prithpal-Sooriya) [inline packages/network-enablement-controller/src/NetworkEnablementController.ts:262]: Nice!
Technically this doesn't do any state manipulation stuff, could be a util or selector?

File: packages/network-enablement-controller/src/constants.ts [hunk 2]
```diff
@@ -9,4 +9,6 @@ export const POPULAR_NETWORKS = [
   '0x89', // Polygon (137)
   '0x531', // Sei (Assuming 1329 used in EVM context)
   '0x144', // zkSync Era (324)
+  '0x2a15c308d', // Palm (11297108109)
+  '0x3e7', // HyperEVM (999)
```
Comment (Prithpal-Sooriya) [inline packages/network-enablement-controller/src/constants.ts:13]: Okie, nice. As you mentioned, we can have this list passed into the controller through a param, this way clients (if need to) can diverge their popular networks.

File: packages/network-enablement-controller/src/NetworkEnablementController.ts (hunk 1 above)
Comment (Prithpal-Sooriya) [reply packages/network-enablement-controller/src/NetworkEnablementController.ts:262]: Not a blocker, we can refactor out if this controller gets really large.
